src/main/java/edu/hawaii/its/api/service/GroupingOwnerService.java

```diff
@@ -84,6 +94,24 @@ public GroupingSyncDestinations groupingsSyncDestinations(String currentUser, St
                 currentUser,
                 findAttributesResults.getResults().stream().map(AttributesResult::getName).collect(Collectors.toList()),
                 groupingPath);
-        return new GroupingSyncDestinations(findAttributesResults, groupAttributeResults);
+
+        GroupingSyncDestinations GroupingSyncDestinations = new GroupingSyncDestinations(findAttributesResults, groupAttributeResults);
+        GroupingSyncDestinations.getSyncDestinations().forEach(e -> e.setDescription((parseKeyVal(groupAttributeResults.getGroups().get(0).getExtension(), e))));
+        GroupingSyncDestinations.getSyncDestinations().sort(Comparator.comparing(GroupingSyncDestination::getDescription));
+        return GroupingSyncDestinations;
+    }
+
+    private String parseKeyVal(String replace, GroupingSyncDestination dest) {
+        String result = replace;
+        if (dest.getName().contains("google-group")) {
+            result += GOOGLE_SYNC_DEST_SUFFIX;
+        }
+
+        try {
+            result = pattern.matcher(dest.getDescription()).replaceFirst(result);
+        } catch (PatternSyntaxException e) {
+            result = dest.getDescription();
+        }
+        return result;
```

Comment:
Could this logic be moved into the GroupingSyncDestinations class? Also see if this bit of code can be refactored to be easier to read, especially the parseKeyVal function.